agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4333,14 +4333,20 @@ bool slave_thread::autoconfig_wsc_parse_m2_encrypted_settings(
 {
     auto encrypted_settings = m2->encrypted_settings();
     uint8_t *iv             = reinterpret_cast<uint8_t *>(encrypted_settings->iv());
-    auto buf                = reinterpret_cast<uint8_t *>(encrypted_settings->encrypted_settings());
+    auto ciphertext         = reinterpret_cast<uint8_t *>(encrypted_settings->encrypted_settings());
+    int clen                = encrypted_settings->encrypted_settings_length();
```

Comment:
The library takes `int` and not `size_t` so isn't it better to do the casting here instead of inside the `aes_decrypt` function?

Reply:
> The library takes `int` and not `size_t` so isn't it better to do the casting here instead of inside the `aes_decrypt` function?

Argh, what a stupid library openssl is... Yes, in that case it has to be int, otherwise passing it as reference becomes problematic. I should have seen that.